### PR TITLE
Pr cc refactoring

### PR DESCRIPTION
**Pull request by fbischoff**

### CODE REVIEW

File: src/madness/chem/CCStructures.h [hunk 1]
```diff
@@ -744,66 +729,25 @@ struct CCConvolutionOperator {
...
-            Parameters param2=param;
-            param2.gamma*=2.0;
-            result.push_back(std::make_pair(1.0*prefactor,CCConvolutionOperator(left.world, OT_ONE, param)));
-            result.push_back(std::make_pair(-2.0*prefactor,CCConvolutionOperator(left.world, OT_SLATER, left.parameters)));
-            result.push_back(std::make_pair(1.0*prefactor,CCConvolutionOperator(left.world, OT_SLATER, param2)));
-        }
-        return result;
+    friend std::shared_ptr<CCConvolutionOperator> combine(const std::shared_ptr<CCConvolutionOperator>& a,
+                                         const std::shared_ptr<CCConvolutionOperator>& b) {
+        auto bla=combine(*a,*b);
```
Comment (kottmanj) [inline src/madness/chem/CCStructures.h:746]: is the "bla" object needed?

File: src/madness/chem/lowrankfunction.h [hunk 2]
```diff
@@ -0,0 +1,587 @@
...
+
+        Function<T,NDIM> reconstruct() const {
+            auto fapprox=hartree_product(g[0],h[0]);
+            for (int i=1; i<g.size(); ++i) fapprox+=hartree_product(g[i],h[i]);
+            return fapprox;
+        }
+
+        /// optimize using Cholesky decomposition
+
+        /// if stable_power_iteration is true, orthonormalize in between applications of the kernel (Alg. 4.4 in Halko)
```
Comment (kottmanj) [inline src/madness/chem/lowrankfunction.h:462]: This one: https://arxiv.org/abs/0909.4061 ?
a bit hard to follow what is happening here.

**Comment by fbischoff** [comment]
This is a functional checkpoint. 

On the way towards using low-rank functions, i.e. a representation f(1,2) = \sum_i g_i(1) h_i(2), and using them in computing MP3 energies and density matrices. 
Long term goal is to have a general function interface class that might either interface to a Function or a low-dimensional representation
